src/slices/requests.ts

```diff
@@ -0,0 +1,60 @@
+import { createSlice, PayloadAction } from '@reduxjs/toolkit'
+import * as api from '../api/requests'
+import { AppThunk } from 'app/store'
+import { RootState } from 'app/rootReducer'
+
+interface IRequestState {
+  isLoading: boolean
+  requests: api.IRequest[]
+  error: string
+}
+
+const initialState: IRequestState = {
+  isLoading: false,
+  requests: [],
+  error: '',
+}
+
+const requestsSlice = createSlice({
+  name: 'requests',
+  initialState,
+  reducers: {
+    getRequests(state) {
+      state = initialState
+    },
+    getRequestsSuccess(state, action: PayloadAction<api.IRequest[]>) {
+      state.isLoading = false
```

Comment:
Notice how we are directly mutating state here, however this operation is completely immutable. This is possible because _redux-toolkit_ behind the scenes uses [_immer.js_](https://immerjs.github.io/immer/docs/introduction). The cool thing is, you do not have to worry about cloning, object/array spreading existing state; just mutate existing state and _immer_ will take care of the rest.